src/dodal/beamlines/i22.py

```diff
@@ -41,338 +39,233 @@
 )
 
 
-def saxs(
-    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> PilatusDetector:
-    return device_instantiation(
-        NXSasPilatus,
-        "saxs",
-        "-EA-PILAT-01:",
-        wait_for_connection,
-        fake_with_ophyd_sim,
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def saxs():
+    """Create a SAXS detector with specific settings."""
+    metadata_holder = NXSasMetadataHolder(
+        x_pixel_size=(1.72e-1, "mm"),
+        y_pixel_size=(1.72e-1, "mm"),
+        description="Dectris Pilatus3 2M",
+        type="Photon Counting Hybrid Pixel",
+        sensor_material="silicon",
+        sensor_thickness=(0.45, "mm"),
+        distance=(4711.833684146172, "mm"),
+    )
+
+    return NXSasPilatus(
+        name="saxs",
+        prefix="-EA-PILAT-01:",
         drv_suffix="CAM:",
         hdf_suffix="HDF5:",
-        metadata_holder=NXSasMetadataHolder(
-            x_pixel_size=(1.72e-1, "mm"),
-            y_pixel_size=(1.72e-1, "mm"),
-            description="Dectris Pilatus3 2M",
-            type="Photon Counting Hybrid Pixel",
-            sensor_material="silicon",
-            sensor_thickness=(0.45, "mm"),
-            distance=(4711.833684146172, "mm"),
-        ),
+        metadata_holder=metadata_holder,
         directory_provider=get_directory_provider(),
     )
 
 
-def synchrotron(
-    wait_for_connection: bool = True, fake_with_ophyd_sim: bool = False
-) -> Synchrotron:
-    return device_instantiation(
-        Synchrotron,
-        "synchrotron",
-        "",
-        wait_for_connection,
-        fake_with_ophyd_sim,
-    )
+@device_factory(default_use_mock_at_connection=True, default_timeout_for_connect=10)
+def synchrotron():
+    """Create a Synchrotron instance with specific settings."""
+    return Synchrotron(name="synchrotron", prefix="")
```

Comment:
See my comment at https://github.com/DiamondLightSource/dodal/pull/597#discussion_r1726675436. I think it would be good to keep some of this stuff here, even if it's not 100% required, otherwise the magic is just not obvious.

Reply:
accounted for this